### ISSUE TITLE
Confusing dead code error unconditionally true '&&' condition

### ISSUE THREAD

**Issue by jez** [sorbet/sorbet#820]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%0Ax%20%3D%20nil%0Ay%20%3D%20nil%0Aif%20x.nil%3F%20%26%26%20y.nil%3F%0A%20%20puts%20x%0Aend)

```ruby
# typed: true

x = nil
y = nil
if x.nil? && y.nil?
  puts x
end
```

#### Observed output

```
editor.rb:5: This code is unreachable https://sorbet.org/docs/error-reference#7006
     5 |if x.nil? && y.nil?
           ^^^^^^^^^^^^^^^^
Errors: 1
```

#### Expected behavior

This condition is unconditionally true (and therefore not dead). That means the dead code error must be coming somewhere with our desugaring of `&&` (or maybe somewhere else).

- - -

**Comment by reitermarkus** [comment]
I found this same problem in an `elsif` branch:

```ruby
only = nil
except = nil

if only
  :only
elsif except
  :except
elsif only.nil? && except.nil?
  :none
end
```

https://sorbet.run/#%23%20typed%3A%20true%0A%0Aonly%20%3D%20nil%0Aexcept%20%3D%20nil%0A%0Aif%20only%0A%20%20%3Aonly%0Aelsif%20except%0A%20%20%3Aexcept%0Aelsif%20only.nil%3F%20%26%26%20except.nil%3F%0A%20%20%3Anone%0Aend%0A

**Comment by jez** [comment]
Fixed in #6139
